Branch-SDK/Branch-SDK/BNCDeviceInfo.m

```diff
@@ -83,27 +83,60 @@ - (id)init {
 
     }
 
-    static NSString* browserUserAgentString = nil;
+    self.browserUserAgent = [self.class userAgentString];
+    return self;
+}
 
-    void (^setUpBrowserUserAgent)() = ^() {
-        browserUserAgentString =
-            [[[UIWebView alloc]
-              initWithFrame:CGRectZero]
-                stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
-        self.browserUserAgent = browserUserAgentString;
-    };
-
-    @synchronized (self.class) {
-        if (browserUserAgentString) {
-            self.browserUserAgent = browserUserAgentString;
-        } else if (NSThread.isMainThread) {
-            setUpBrowserUserAgent();
-        } else {
-            dispatch_sync(dispatch_get_main_queue(), setUpBrowserUserAgent);
-        }
-    }
++ (NSString*) userAgentString {
 
-    return self;
+    static NSString* browserUserAgentString = nil;
+	void (^setBrowserUserAgent)() = ^() {
+		if (!browserUserAgentString) {
+			browserUserAgentString =
+				[[[UIWebView alloc]
+				  initWithFrame:CGRectZero]
+					stringByEvaluatingJavaScriptFromString:@"navigator.userAgent"];
+			NSLog(@"[Branch] userAgentString: '%@'.", browserUserAgentString);
```

Comment:
I'll remove it after the final QA round.  It'll be much easier to see if this is breaking with the logging in.